### PR TITLE
Belt single asset strategy

### PR DESCRIPTION
**Pull request by canner1**
Belt.fi allows users to deposit BTC, ETH, WBNB tokens to their liquidity pool and stake the share tokens. 3 beltToken staking pools are added, beltBTC pool with 591.2m tvl, beltETH pool with 525.3m tvl and beltBNB pool with 244m tvl.

Test results: `test/belt/beltBnb.js`
```
  BSC Mainnet BeltBNB
Impersonating...
0xf00dD244228F51547f0563e60bCa65a30FBF5f7f
Fetching Underlying at:  0xa8Bb71facdd46445644C277F9499Dd22f6F0A30C
New Vault Deployed:  0x4C43d8C3ec92047c2c24775ad4665894930FBfF5
Strategy Deployed:  0x9E545E3C0baAB3E08CdfD552C960A1050f373042
Strategy and vault added to Controller.
    Happy path
loop  0
old shareprice:  1000000000000000000
new shareprice:  1000000000000000000
growth:  1
loop  1
old shareprice:  1000000000000000000
new shareprice:  1000029457787319427
growth:  1.0000294577873194
loop  2
old shareprice:  1000029457787319427
new shareprice:  1000058916289575070
growth:  1.0000294576344988
loop  3
old shareprice:  1000058916289575070
new shareprice:  1000088375506681072
growth:  1.0000294574815805
loop  4
old shareprice:  1000088375506681072
new shareprice:  1000117835438740785
growth:  1.0000294573287534
loop  5
old shareprice:  1000117835438740785
new shareprice:  1000147296085762959
growth:  1.000029457175923
loop  6
old shareprice:  1000147296085762959
new shareprice:  1000176757447661727
growth:  1.0000294570229946
loop  7
old shareprice:  1000176757447661727
new shareprice:  1000206219524540449
growth:  1.0000294568701575
loop  8
old shareprice:  1000206219524540449
new shareprice:  1000235682316407872
growth:  1.0000294567173171
loop  9
old shareprice:  1000235682316407872
new shareprice:  1000265145823178123
growth:  1.000029456564379
earned!
APR: 12.903919969010103 %
APY: 13.77087771099843 %
      ✓ Farmer should earn money (131414ms)
```

`test/belt/beltEth.js`:
```
  BSC Mainnet BeltETH
Impersonating...
0xf00dD244228F51547f0563e60bCa65a30FBF5f7f
Fetching Underlying at:  0xAA20E8Cb61299df2357561C2AC2e1172bC68bc25
New Vault Deployed:  0x4C43d8C3ec92047c2c24775ad4665894930FBfF5
Strategy Deployed:  0x9E545E3C0baAB3E08CdfD552C960A1050f373042
Strategy and vault added to Controller.
    Happy path
loop  0
old shareprice:  1000000000000000000
new shareprice:  1000000000000000000
growth:  1
loop  1
old shareprice:  1000000000000000000
new shareprice:  1000013466803527685
growth:  1.0000134668035277
loop  2
old shareprice:  1000013466803527685
new shareprice:  1000026933752644985
growth:  1.000013466767763
loop  3
old shareprice:  1000026933752644985
new shareprice:  1000040400847335577
growth:  1.0000134667319809
loop  4
old shareprice:  1000040400847335577
new shareprice:  1000053868087617101
growth:  1.0000134666962157
loop  5
old shareprice:  1000053868087617101
new shareprice:  1000067335473490215
growth:  1.0000134666604499
loop  6
old shareprice:  1000067335473490215
new shareprice:  1000080803004955577
growth:  1.0000134666246838
loop  7
old shareprice:  1000080803004955577
new shareprice:  1000094270681996863
growth:  1.0000134665889004
loop  8
old shareprice:  1000094270681996863
new shareprice:  1000107738504631715
growth:  1.0000134665531337
loop  9
old shareprice:  1000107738504631715
new shareprice:  1000121206472860789
growth:  1.0000134665173668
earned!
APR: 5.898746896794815 %
APY: 6.075689268395323 %
      ✓ Farmer should earn money (123870ms)
```

`test/belt/beltBtc.js`:

```
  BSC Mainnet BeltBTC
Impersonating...
0xf00dD244228F51547f0563e60bCa65a30FBF5f7f
Fetching Underlying at:  0x51bd63F240fB13870550423D208452cA87c44444
New Vault Deployed:  0x4C43d8C3ec92047c2c24775ad4665894930FBfF5
Strategy Deployed:  0x9E545E3C0baAB3E08CdfD552C960A1050f373042
Strategy and vault added to Controller.
    Happy path
loop  0
old shareprice:  1000000000000000000
new shareprice:  1000000000000000000
growth:  1
loop  1
old shareprice:  1000000000000000000
new shareprice:  1000024268943173099
growth:  1.000024268943173
loop  2
old shareprice:  1000024268943173099
new shareprice:  1000048538359235932
growth:  1.0000242688270842
loop  3
old shareprice:  1000048538359235932
new shareprice:  1000072808248192358
growth:  1.000024268710993
loop  4
old shareprice:  1000072808248192358
new shareprice:  1000097078610046237
growth:  1.0000242685949
loop  5
old shareprice:  1000097078610046237
new shareprice:  1000121349444800362
growth:  1.0000242684788039
loop  6
old shareprice:  1000121349444800362
new shareprice:  1000145620752459657
growth:  1.000024268362707
loop  7
old shareprice:  1000145620752459657
new shareprice:  1000169892533027978
growth:  1.000024268246608
loop  8
old shareprice:  1000169892533027978
new shareprice:  1000194164786509184
growth:  1.0000242681305072
loop  9
old shareprice:  1000194164786509184
new shareprice:  1000218437512907132
growth:  1.000024268014404
earned!
APR: 10.630729195490618 %
APY: 11.214636882671304 %
      ✓ Farmer should earn money (107013ms)
```

### CODE REVIEW

File: contracts/strategies/belt/BeltSingleAssetStrategy.sol [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+import "@pancakeswap/pancake-swap-lib/contracts/math/SafeMath.sol";
+
+import "../../base/upgradability/BaseUpgradeableStrategy.sol";
+import "../../base/interface/pancakeswap/IPancakeRouter02.sol";
+
+contract BeltSingleAssetStrategy is BaseUpgradeableStrategy {
+  using SafeMath for uint256;
+  using SafeBEP20 for IBEP20;
+
+  address constant public pancakeswapRouterV2 = address(0x05fF2B0DB69458A0750badebc4f9e13aDd608C7F);
```
Comment (CryptJS13) [inline contracts/strategies/belt/BeltSingleAssetStrategy.sol:21]: Use new Pancake router: 0x10ED43C718714eb63d5aA57B78B54704E256024E

File: test/belt/beltBtc.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+const { send } = require("@openzeppelin/test-helpers");
+const BigNumber = require("bignumber.js");
+const IBEP20 = artifacts.require("IBEP20");
+const IMultiStrategyToken = artifacts.require("IMultiStrategyToken");
+const Strategy = artifacts.require("BeltSingleAssetStrategyMainnet_BeltBTCB");
+
+
+// Vanilla Mocha test. Increased compatibility with tools that integrate Mocha.
+describe("BSC Mainnet BeltBSC", function() {
```
Comment (CryptJS13) [inline test/belt/beltBtc.js:18]: Typo, BeltBTC

File: contracts/strategies/belt/BeltSingleAssetStrategy.sol (hunk 1 above)
Comment (deghub) [reply contracts/strategies/belt/BeltSingleAssetStrategy.sol]: resolved

File: test/belt/beltBtc.js (hunk 2 above)
Comment (deghub) [reply test/belt/beltBtc.js]: fixed

**Review by breadforthepeople — APPROVED** [review]

**Pull request merged**